python/ray/train/predictor.py

```diff
@@ -124,6 +124,32 @@ def set_preprocessor(self, preprocessor: Optional[Preprocessor]) -> None:
         """Set the preprocessor to use prior to executing predictions."""
         self._preprocessor = preprocessor
 
+    @classmethod
+    @DeveloperAPI
+    def preferred_batch_format(cls) -> BatchFormat:
+        """Batch format hint for upstream producers to try yielding best block format.
+        Can be overriden by predictor classes depending on the framework type.
+
+        1. In case for DLPredictor that requires Numpy for neural network inference,
+        this method should return BatchFormat.NUMPY.
+        2. In case XGBoostPredictor that primarily works with tabular data, this
+        method can return BatchFormat.PANDAS.
```

Comment:
the `_predict_pandas` and `_predict_numpy` are part of `_batch_format_to_use`'s logic rather than this one, it's more of an example of pandas / numpy native predictors